### PR TITLE
Fix: canary not running

### PR DESCRIPTION
**Pull request by markkuhn**
*Description of changes*

Canary hasn't been running since https://github.com/awslabs/aws-embedded-metrics-java/commit/fba1075e4fadc7e5ac094e642c74f5ddd046c553.
The execution waits for `environmentFuture.join()` indefinitely in the MetricsLogger flusher.

This is similar to #95 but I cannot yet confirm this PR will resolve it.

The change I have made is to replace all `org.javatuples.Pair` in the environments package with `java.util.Map` since calling `new Pair<>()` gets the whole thread parked indefinitely. It might be a bug in javatuples or we might not be using it properly, this I am unsure of.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by gordonpn — APPROVED** [review]

File: canarytests/agent/src/main/java/emf/canary/ECSRunnable.java [hunk 1]
```diff
@@ -13,14 +14,17 @@ public class ECSRunnable implements Runnable {
 
     @Override
     public void run() {
+        final Configuration config = EnvironmentConfigurationProvider.getConfig();
+        config.setEnvironmentOverride(Environments.Agent);
```
Comment (gordonpn) [inline canarytests/agent/src/main/java/emf/canary/ECSRunnable.java:18]: What does this do?

Comment (markkuhn) [reply canarytests/agent/src/main/java/emf/canary/ECSRunnable.java:18]: This is a temporary override to skip environment detection. I was using it to test the canary. Will remove.

**Comment by markkuhn** [comment]
Canary successfully running [here](https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#dashboards:name=emf-java-deployment).

File: canarytests/agent/src/main/java/emf/canary/ECSRunnable.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import software.amazon.cloudwatchlogs.emf.config.Configuration;
 import software.amazon.cloudwatchlogs.emf.config.EnvironmentConfigurationProvider;
+import software.amazon.cloudwatchlogs.emf.environment.Environments;
```
Comment (gordonpn) [inline canarytests/agent/src/main/java/emf/canary/ECSRunnable.java:5]: I don't think this is being used anymore.

**Review by gordonpn — APPROVED** [review]

**Review by evanuk — APPROVED** [review]

**Review by Himtanaya — APPROVED** [review]

**Comment by markkuhn** [comment]
Update: Root cause was `org.javatuples.Pair` not being present in the `build.gradle` file in `canarytests`. This issue was resolves by using `Map` instead since it wasn't relying on a non-existent package.

**Pull request merged**